packages/datasource-hdt/lib/datasources/HdtDatasource.js

```diff
@@ -48,8 +51,11 @@ class HdtDatasource extends Datasource {
           estimatedTotalCount = offset + (tripleCount < query.limit ? tripleCount : 2 * tripleCount);
         destination.setProperty('metadata', { totalCount: estimatedTotalCount, hasExactCount: hasExactCount });
         // Add the triples to the output
-        for (var i = 0; i < tripleCount; i++)
-          destination._push(triples[i]);
+        for (var i = 0; i < tripleCount; i++) {
+          let triple = RdfString.stringQuadToQuad(triples[i], dataFactory);
+          delete triple.graph; // Triples in a HDT do not contain a graph.
```

Comment:
Is this graph field removal necessary?
Because this breaks with the RDFJS quad contract, which may lead to crashes downstream.
I guess this just contains the default graph, right?

Reply:
This indeed only contains the default graph. 
I deleted this because the tests request no graph to be present, in case that there is no custom graph set for the datasource.
But I will remove this and the one in the test file, as the absence of a graph field indicates that a triple is part of the default graph in this context anyway.